pkg/cimfs/cim_writer_windows.go

```diff
@@ -289,3 +369,47 @@ func GetCimUsage(ctx context.Context, cimPath string) (uint64, error) {
 	}
 	return totalUsage, nil
 }
+
+// MergeBlockCIMs creates a new merged BlockCIM from the provided source BlockCIMs.  CIM
+// at index 0 is considered to be topmost CIM and the CIM at index `length-1` is
+// considered the base CIM. (i.e file with the same path in CIM at index 0 will shadow
+// files with the same path at all other CIMs)
+// When mounting this merged CIM the source CIMs MUST be provided in the exact same order.
+func MergeBlockCIMs(mergedCIM *BlockCIM, sourceCIMs []*BlockCIM) (err error) {
```

Comment:
Do we only support merging block cims?

Reply:
Would it be possible in the future to merge file cims and block cims together?

Reply:
Isn't the point of `pkg/cimfs` to expose generic CimFS capabilities, though, rather than scoped to only our needs? If it's not too hard to expose something more generic, that seems like a good path to me.

Reply:
As of now merging of standard (i.e non-block CIMs) isn't supported. We will add the corresponding wrapper in future if required.